agent/api/container/container.go

```diff
@@ -925,3 +928,40 @@ func (c *Container) AddContainerDependency(name string, condition string) {
 		Condition:     condition,
 	})
 }
+
+// GetLogDriver returns the log driver used by the container.
+func (c *Container) GetLogDriver() string {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	if c.DockerConfig.HostConfig == nil {
+		return ""
+	}
+
+	hostConfig := &dockercontainer.HostConfig{}
+	err := json.Unmarshal([]byte(*c.DockerConfig.HostConfig), hostConfig)
+	if err != nil {
+		seelog.Warnf("Encounter error when trying to get log driver for container %s: %v", err)
+		return ""
+	}
+
+	return hostConfig.LogConfig.Type
+}
+
+// HasSecret returns whether a container has a secret with a certain provider and target type.
```

Comment:
We already have a helper function HasSecretAsEnvOrLogDriver, it would be better to combine these two functions since both of them are used for checking secret type or target.

Reply:
but these two are doing different things.. i'm not sure how to combine them?